server/schedule.py

```diff
@@ -23,9 +53,23 @@ def _store(self):
         pass
 
     # stores everything on given date
-    def _update_date(self, date):
-        pass
+    def _update_date(self, timestamp):
+        f = open(self.directory + str(timestamp) + ".txt", "w")
+        f.writelines(self.mem_sched[timestamp])
+        f.close()
 
     # appends just 1 item
-    def _append_to_date(self, uid, date):
-        pass
+    def _append_to_date(self, uid, timestamp):
+        f = open(self.directory + str(timestamp) + ".txt", "a")
+        f.write(uid + '\n')
+        f.close()
+
+    def _read_file(self, timestamp) -> list:
+        # load in from disc
+        filename = self.directory + str(timestamp) + ".txt"
+        # create file if doesn't exist (a+)
+        with open(filename, "a+") as f:
```

Comment:
a+ is for appending, but you then seek(0), I think r+ will seek(0) by default

Reply:
@ConorMurphy21 i've heard that r+ doesnt create a file if it doesn't exist. a lot of the research I did didn't really clarify. can you briefly look into it?

Reply:
you are reading anyways, if it doesn't create a file that means there is nothing on disk. So just set your return value above the with to be empty?